src/v/raft/tests/stm_test_fixture.h

```diff
@@ -168,7 +168,7 @@ struct state_machine_fixture : raft_fixture {
           [b = std::move(builder).build()](
             raft_node_instance& leader_node) mutable {
               return leader_node.raft()->replicate(
-                model::make_memory_record_batch_reader(std::move(b)),
+                model::make_memory_record_batch_reader(b.share()),
```

Comment:
@bharathv why was this a use-after-move? from what i can tell, the std::move was the last reference to `b`

Reply:
oh i see the retry will invoke the lambda more than once